pkg/pusher/pusher.go

```diff
@@ -165,21 +167,27 @@ LOOP:
 						// this is to make sure that the sent number does not diverge from the synced counter
 						// the edge case is on the uploader node, in the case where the uploader node is
 						// connected to other nodes, but is the closest one to the chunk.
-						setSent = true
+						wantSelf = true
 					} else {
 						return
 					}
 				}
 
-				publicKey, err := crypto.Recover(receipt.Signature, receipt.Address.Bytes())
-				if err != nil {
-					err = fmt.Errorf("pusher: receipt recover: %w", err)
-					return
-				}
+				if receipt != nil {
+					var publicKey *ecdsa.PublicKey
+					publicKey, err = crypto.Recover(receipt.Signature, receipt.Address.Bytes())
+					if err != nil {
+						err = fmt.Errorf("pusher: receipt recover: %w", err)
+						return
+					}
 
-				storerPeer, err = crypto.NewOverlayAddress(*publicKey, s.networkID)
-				if err != nil {
-					err = fmt.Errorf("pusher: receipt storer address: %w", err)
+					storerPeer, err = crypto.NewOverlayAddress(*publicKey, s.networkID)
+					if err != nil {
+						err = fmt.Errorf("pusher: receipt storer address: %w", err)
+						return
+					}
+				} else if !wantSelf { // receipt is nil, so ErrWantSelf had to be returned
```

Comment:
not sure if it's necessary to check this. if I understand correctly, this case is `receipt == nil && !wantSelf` which shouldn't happen at all no?

Reply:
this is technically a part of the signature verification process. We should always get a receipt, and it should be signed. I think the added check is a good thing for cases where people run nodes with modified code bases. What do you think?

Reply:
ahh, the receipt is returned from PushChunkToClosest, the origin node constructs it from the stream, so yes, the extra check is not needed.